crates/router/src/connector/klarna/transformers.rs

```diff
@@ -157,7 +148,9 @@ impl TryFrom<&KlarnaRouterData<&types::PaymentsSessionRouterData>> for KlarnaSes
                         name: data.product_name.clone(),
                         quantity: data.quantity,
                         unit_price: data.amount,
-                        total_amount: i64::from(data.quantity) * (data.amount),
+                        total_amount: MinorUnit::new(
+                            i64::from(data.quantity) * (data.amount.get_amount_as_i64()),
```

Comment:
we shouldn't be using the func `get_amount_as_i64()` one way to approach this problem is to make 
```suggestion
                            MinorUnit::new(i64::from(data.quantity)) * (data.amount),
```
cc: @jarnura